brave/src/main/java/brave/internal/recorder/PendingSpans.java

```diff
@@ -120,21 +120,22 @@ void reportOrphanedSpans() {
     // flushing a span than hurt performance of unrelated operations by calling
     // currentTimeMicroseconds N times
     long flushTime = 0L;
-    boolean noop = zipkinHandler == FinishedSpanHandler.NOOP || this.noop.get();
+    boolean noop = orphanedSpanHandler == FinishedSpanHandler.NOOP || this.noop.get();
     while ((contextKey = (RealKey) poll()) != null) {
       PendingSpan value = delegate.remove(contextKey);
       if (noop || value == null || !contextKey.sampled) continue;
       if (flushTime == 0L) flushTime = clock.currentTimeMicroseconds();
 
+      boolean isEmpty = value.state.isEmpty();
+      Throwable caller = value.caller;
+
       TraceContext context = InternalPropagation.instance.newTraceContext(
         InternalPropagation.FLAG_SAMPLED_SET | InternalPropagation.FLAG_SAMPLED,
         contextKey.traceIdHigh, contextKey.traceId,
         contextKey.localRootId, 0L, contextKey.spanId,
-        Collections.emptyList()
+        caller != null && !isEmpty ? Collections.singletonList(caller) : Collections.emptyList()
```

Comment:
untested change here is adding the calling stack trace into the context, so that a finished span handler could processes it in a way different than logging to hunt down who created the bug.